### ISSUE TITLE
Support UDP NAT type 2 in dokodemo-door

### ISSUE THREAD

**Issue by DarienRaymond** [v2ray/Planning#45]
So far it is type 4 (symmetric) due to some internal designs.

It requires fundamental changes to both internal data structure and VMess protocol to support type 2.

**Comment by xiaokangwang** [comment]
Since we needs to change VMess protocol to support type 2, why not consider to also amend vmess so that udp traffic can be transferred with udp and decrease delay.

**Comment by DarienRaymond** [comment]
No. VMess is designed to be TCP only. If you need a protocol for UDP transfer, we can design another protocol.

**Comment by xiaokangwang** [comment]
OK, I suppose you mean vmess is based on stream transport.

**Comment by LiangkeZhong** [comment]
希望v2ray的vmess能支持UDP转发。
I hope vmess can support UDP traffic.

**Comment by DarienRaymond** [comment]
一直都支持啊

**Comment by LiangkeZhong** [comment]
但是我这边用supersocks5却不行，提示UDP被封锁，同等情况下ssr和ss就不会。

**Comment by DarienRaymond** [comment]
请在 [v2ray-core](https://github.com/v2ray/v2ray-core) 开一个 issue
